### ISSUE TITLE
setup_requires causes issues on Python 2.6

### ISSUE THREAD

**Issue by moggers87** [carljm/django-model-utils#179]
If I try to install Django and django-model-utils at the same time, a second copy of Django gets downloaded to satisfy the `setup_requires` line in `setup.py`. This won't normally cause issues, but on Python 2.6 it will produce the following:

```
$ pip install -r test-requires.txt 
Downloading/unpacking Django<1.7 (from -r test-requires.txt (line 1))
  Downloading Django-1.6.11.tar.gz (6.8MB): 6.8MB downloaded
  Running setup.py egg_info for package Django
    warning: no previously-included files matching '__pycache__' found under directory '*'
    warning: no previously-included files matching '*.py[co]' found under directory '*'
Downloading/unpacking django-model-utils (from -r test-requires.txt (line 2))
  Downloading django-model-utils-2.3.tar.gz
  Running setup.py egg_info for package django-model-utils
    Traceback (most recent call last):
      File "<string>", line 16, in <module>
      File "/tmp/pip-build-moggers87/django-model-utils/setup.py", line 92, in <module>
        'compile_translations': compile_translations}
      File "/usr/lib/python2.6/distutils/core.py", line 113, in setup
        _setup_distribution = dist = klass(attrs)
      File "/usr/lib/python2.6/site-packages/setuptools/dist.py", line 221, in __init__
        self.fetch_build_eggs(attrs.pop('setup_requires'))
      File "/usr/lib/python2.6/site-packages/setuptools/dist.py", line 245, in fetch_build_eggs
        parse_requirements(requires), installer=self.fetch_build_egg
      File "/usr/lib/python2.6/site-packages/pkg_resources.py", line 538, in resolve
        dist = best[req.key] = env.best_match(req, self, installer)
      File "/usr/lib/python2.6/site-packages/pkg_resources.py", line 780, in best_match
        return self.obtain(req, installer) # try and download/install
      File "/usr/lib/python2.6/site-packages/pkg_resources.py", line 792, in obtain
        return installer(requirement)
      File "/usr/lib/python2.6/site-packages/setuptools/dist.py", line 293, in fetch_build_egg
        return cmd.easy_install(req)
      File "/usr/lib/python2.6/site-packages/setuptools/command/easy_install.py", line 472, in easy_install
        return self.install_item(spec, dist.location, tmpdir, deps)
      File "/usr/lib/python2.6/site-packages/setuptools/command/easy_install.py", line 502, in install_item
        dists = self.install_eggs(spec, download, tmpdir)
      File "/usr/lib/python2.6/site-packages/setuptools/command/easy_install.py", line 681, in install_eggs
        return self.build_and_install(setup_script, setup_base)
      File "/usr/lib/python2.6/site-packages/setuptools/command/easy_install.py", line 958, in build_and_install
        self.run_setup(setup_script, setup_base, args)
      File "/usr/lib/python2.6/site-packages/setuptools/command/easy_install.py", line 947, in run_setup
        run_setup(setup_script, args)
      File "/usr/lib/python2.6/site-packages/setuptools/sandbox.py", line 29, in run_setup
        lambda: execfile(
      File "/usr/lib/python2.6/site-packages/setuptools/sandbox.py", line 70, in run
        return func()
      File "/usr/lib/python2.6/site-packages/setuptools/sandbox.py", line 31, in <lambda>
        {'__file__':setup_script, '__name__':'__main__'}
      File "setup.py", line 32, in <module>
    
      File "/tmp/easy_install-kQyYF7/Django-1.8.3/django/__init__.py", line 1, in <module>
        #
      File "/tmp/easy_install-kQyYF7/Django-1.8.3/django/utils/version.py", line 7, in <module>
      File "/tmp/easy_install-kQyYF7/Django-1.8.3/django/utils/lru_cache.py", line 28
        fasttypes = {int, str, frozenset, type(None)},
                        ^
    SyntaxError: invalid syntax
```

test-requires.txt for completeness:
```
Django<1.7
django-model-utils
```

I guess the setup_requires could be tighten if we're on Python 2.6

**Comment by rafaduran** [comment]
I'm also getting issues, even if the right Django version is installed first:

```
(26)apollo:ntteam.commons rduran$ pip install "Django<1.7"
You are using pip version 6.1.1, however version 7.1.0 is available.
You should consider upgrading via the 'pip install --upgrade pip' command.
Collecting Django<1.7
  Using cached Django-1.6.11-py2.py3-none-any.whl
Installing collected packages: Django
Successfully installed Django-1.6.11
(26)apollo:ntteam.commons rduran$ pip install --pre django-model-utils
You are using pip version 6.1.1, however version 7.1.0 is available.
You should consider upgrading via the 'pip install --upgrade pip' command.
Collecting django-model-utils
  Using cached django-model-utils-2.3.tar.gz
Requirement already satisfied (use --upgrade to upgrade): Django>=1.4.2 in /Users/rduran/.virtualenvs/26/lib/python2.6/site-packages (from django-model-utils)
Installing collected packages: django-model-utils
  Running setup.py install for django-model-utils
    Complete output from command /Users/rduran/.virtualenvs/26/bin/python2.6 -c "import setuptools, tokenize;__file__='/private/var/folders/yv/8yfdn6v57flf56ffmrbh55340000gn/T/pip-build-K6Enkz/django-model-utils/setup.py';exec(compile(getattr(tokenize, 'open', open)(__file__).read().replace('\r\n', '\n'), __file__, 'exec'))" install --record /var/folders/yv/8yfdn6v57flf56ffmrbh55340000gn/T/pip-oJPH8d-record/install-record.txt --single-version-externally-managed --compile --install-headers /Users/rduran/.virtualenvs/26/include/site/python2.6/django-model-utils:
    running install
    running build
    running compile_translations
    CommandError: Can't find msgfmt. Make sure you have GNU gettext tools 0.15 or newer installed.
    
    ----------------------------------------
    Command "/Users/rduran/.virtualenvs/26/bin/python2.6 -c "import setuptools, tokenize;__file__='/private/var/folders/yv/8yfdn6v57flf56ffmrbh55340000gn/T/pip-build-K6Enkz/django-model-utils/setup.py';exec(compile(getattr(tokenize, 'open', open)(__file__).read().replace('\r\n', '\n'), __file__, 'exec'))" install --record /var/folders/yv/8yfdn6v57flf56ffmrbh55340000gn/T/pip-oJPH8d-record/install-record.txt --single-version-externally-managed --compile --install-headers /Users/rduran/.virtualenvs/26/include/site/python2.6/django-model-utils" failed with error code 1 in /private/var/folders/yv/8yfdn6v57flf56ffmrbh55340000gn/T/pip-build-K6Enkz/django-model-utils
```

**Comment by moggers87** [comment]
@rafaduran I think your issue is #178

**Comment by rafaduran** [comment]
@moggers87 yes it is, I think both issues are actually the same.

**Comment by carljm** [comment]
I just pushed a commit to fix this, and will issue a 2.3.1 release. Sorry everyone for the hassle of the broken release; my fault for not reviewing that PR closely enough to realize the implications of the `setup.py` changes.

**Comment by carljm** [comment]
2.3.1 is released. Sorry again for the breakage.
